src/runtime/recipe/handle.ts

```diff
@@ -223,9 +223,14 @@ export class Handle implements Comparable<Handle> {
     return TypeChecker.processTypeList(handleType ? handleType._cloneWithResolutions(variableMap) : null, typeSet);
   }
 
-  static resolveEffectiveType(handleType: Type, connections: HandleConnection[]): Type {
-    const typeSet = connections.filter(connection => connection.type != null).map(connection => ({type: connection.type, direction: connection.direction}));
-    return TypeChecker.processTypeList(handleType, typeSet);
+  static resolveEffectiveType(handleType: Type, connections: HandleConnection[], options: IsValidOptions): Type {
+    const typeSet: TypeListInfo[] = connections.filter(
+      connection => connection.type != null
+    ).map(
+    connection => (
+      {type: connection.type, direction: connection.direction}
+    ));
```

Comment:
Line breaks look kind of strange here. I think you can condense these quite a bit.

Reply:
I agree; either go back to a single line, or
```
const typeSet = connections
    .filter(connection => connection.type != null)
    .map(connection => {type: connection.type, direction: connection.direction}));
```

What you have currently is too hard to read.

Reply:
If I can push back a little, if this was written with the equivalent procedural code, I would find it hard to read if it were condensed down to only one or two lines.
I think is one of the problems often cited when pushing against 'point free' style in functional languages.

I think your formatting, Shane, is pretty neat.
Is a single or double indent expected?